src/twisted/internet/test/test_unix.py

```diff
@@ -1014,3 +1016,104 @@ def connect(self, reactor, factory):
 
 
 globals().update(UnixClientTestsBuilder.makeTestCaseClasses())
+
+
+@skipIf(platformType != "posix", "UNIX sockets only supported on POSIX.")
+class RealSocketTests(TestCase):
+    """
+    Test real UNIX socket connections.
+    """
+
+    path = "unixpath"
+
+    def test_closePortInProtocolFactory(self):
```

Comment:
New tests should really have type annotations.
```suggestion
    def test_closePortInProtocolFactory(self) -> None:
```